sklearn_numba_dpex/kmeans/kernels/lloyd_single_step.py

```diff
@@ -111,23 +111,10 @@ def make_lloyd_single_step_fixed_window_kernel(
 
     n_subgroups = math.ceil(n_samples / window_n_centroids)
 
-    n_cluster_items = n_clusters * (n_features + 1)
-    n_cluster_bytes = np.dtype(dtype).itemsize * n_cluster_items
-    # TODO: control that this value is not higher than the number of sub-groups of size
-    # sub_group_size that can effectively run concurrently. We should fetch this
-    # information and apply it here.
-
     # NB: for more details about the privatization strategy the following variables
     # refer too, please read the inline commenting that address it in the kernel
     # definition.
 
-    # Ensure that the memory allocated for privatization will not saturate the global
-    # memory cache size. For this purpose we limit the number of private copies to
-    # a fraction of the available global memory cache size.
-    n_centroids_private_copies = (
-        global_mem_cache_size * centroids_private_copies_max_cache_occupancy
-    ) // n_cluster_bytes
```

Comment:
Experiments have shown that having a too low number of private copies results in catastrophic performance. Compared to that, saturating memory caches is not much of a worry.